web/src/user_events.js

```diff
@@ -128,6 +129,38 @@ export const update_person = function update(person) {
 
     if (Object.hasOwn(person, "custom_profile_field")) {
         people.set_custom_profile_field_data(person.user_id, person.custom_profile_field);
+        const $navbar_alert = $("#navbar_alerts_wrapper").children(".alert").first();
+        if (
+            !$navbar_alert?.length ||
+            $navbar_alert.is("#empty-required-profile-fields-warning") ||
+            $navbar_alert.is(":hidden")
+        ) {
+            navbar_alerts.maybe_show_empty_required_profile_fields_alert();
+        }
+
+        const field_id = person.custom_profile_field.id;
+        const field_value = people.get_custom_profile_data(person.user_id, field_id)?.value;
+        const is_field_required = realm.custom_profile_fields?.find(
+            (f) => field_id === f.id,
+        )?.required;
+        if (is_field_required) {
+            const $custom_user_field = $(
+                `.custom_user_field[data-field-id="${CSS.escape(field_id)}"]`,
+            );
+            const $field = $custom_user_field.find(".field");
+            const $required_symbol = $custom_user_field.find(".required-symbol");
+            if (!field_value) {
+                if (!$field.hasClass("empty-required-field")) {
+                    $field.addClass("empty-required-field");
+                    $required_symbol.removeClass("hidden");
+                }
+            } else {
+                if ($field.hasClass("empty-required-field")) {
+                    $field.removeClass("empty-required-field");
+                    $required_symbol.addClass("hidden");
+                }
+            }
+        }
```

Comment:
The rest of this logic also seems to assume that the user whose data was edited is the current user. This is the kind of bug that's very hard to catch while testing (though testing live-update with another user account is highly recommended and might catch this) -- you really need to always be thinking when writing logic "what is the exact condition for when this should happen?".

Reply:
I thought this block of code is actually needed even if the fields that are updated are of another user in the "Manage user" tab of the user profile modal, but the user profile modal closes after changes are saved. So it is redundant for that use case right now, but there is an open issue ([#26692](https://github.com/zulip/zulip/issues/26692)) that might change this in the future. I am putting it in ```person.user_id === people.my_current_user_id()``` condition for now though.

Reply:
Just to be concrete, is this the case where you as an administrator are editing a "required" field for another user via their user profile?

I guess I'm not sure how the styling for that should work -- does this colored outline apply in that case? I guess that's reasonable, but I think even once we do #26692, I'd expect us to want to handle these updates by just rerendering the profile.

But what this discussion does cause me to realize is that your selector, `.custom_user_field[data-field-id="${CSS.escape(field_id)}"]`, is insufficiently precise -- will that change the styling on __someone else's__ profile if one of your custom profile fields is edited in another window?

Reply:
Oh, you are right. I missed that. I have fixed it now.